app/ts/McdaApp/Workspace/CurrentTab/Preferences/EquivalentChange/EquivalentChangeContext/EquivalentChangeContext.tsx

```diff
@@ -32,147 +32,229 @@ export function EquivalentChangeContextProviderComponent({
       _.every(pvfs, ['type', 'linear']) &&
       !_.isEmpty(observedRanges),
-    [currentScenario, observedRanges, pvfs]
+    [areAllPvfsSet, currentScenario.state?.weights, observedRanges, pvfs]
   );
 
-  function reset(): void {
-    if (areAllPvfsSet) {
-      const [configuredLowerBound, configuredUpperBound] = getBounds(
-        referenceCriterion.dataSources[0].id,
+  useEffect(() => {
+    if (canShowEquivalentChanges) {
+      const newReferenceCriterion =
+        _.find(filteredCriteria, [
+          'id',
+          currentScenario.state.equivalentChange?.referenceCriterionId
+        ]) || filteredCriteria[0];
+      const newOtherCriteria = _.reject(filteredCriteria, [
+        'id',
+        newReferenceCriterion.id
+      ]);
+      const [initialLowerBound, initialUpperBound] = getBounds(
+        newReferenceCriterion.dataSources[0].id,
         currentSubproblem.definition.ranges,
         observedRanges
       );
-      setLowerBound(configuredLowerBound);
-      setUpperBound(configuredUpperBound);
-      setReferenceValueBy(
-        getInitialReferenceValueBy(configuredLowerBound, configuredUpperBound)
+      const newReferenceValueBy = currentScenario.state.equivalentChange
+        ?.referenceValueBy
+        ? currentScenario.state.equivalentChange.referenceValueBy
+        : getInitialReferenceValueBy(initialLowerBound, initialUpperBound);
+      const newReferencevalueFrom = currentScenario.state.equivalentChange
+        ?.referenceValueFrom
+        ? currentScenario.state.equivalentChange.referenceValueFrom
+        : getInitialReferenceValueFrom(
+            initialLowerBound,
+            initialUpperBound,
+            pvfs[newReferenceCriterion.id]
+          );
+      const newReferenceValueTo = currentScenario.state.equivalentChange
+        ?.referenceValueTo
+        ? currentScenario.state.equivalentChange.referenceValueTo
+        : getInitialReferenceValueTo(
+            initialLowerBound,
+            initialUpperBound,
+            pvfs[newReferenceCriterion.id]
+          );
+      const newPartOfInterval = getPartOfInterval(
+        0,
+        newReferenceValueBy,
+        initialLowerBound,
+        initialUpperBound
       );
-      setReferenceValueFrom(
-        getInitialReferenceValueFrom(
-          configuredLowerBound,
-          configuredUpperBound,
-          pvfs[referenceCriterion.id]
-        )
+      setReferenceCriterion(newReferenceCriterion);
+      setOtherCriteria(newOtherCriteria);
+      setLowerBound(initialLowerBound);
+      setUpperBound(initialUpperBound);
+      setReferenceValueBy(newReferenceValueBy);
+      setReferenceValueFrom(newReferencevalueFrom);
+      setReferenceValueTo(newReferenceValueTo);
+      setPartOfInterval(newPartOfInterval);
+    }
+    // currentScenario is not in here because I don't want it to trigger this
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    canShowEquivalentChanges,
```

Comment:
I would prefer to have the `equivalentChange` object available from the CurrentScenarioContext similar to how we handle pvfs; that would remove the direct circular dependency in this useeffect if i understand it correctly